### PR TITLE
Increase Coverage to ClusterPipeline and Responses classes

### PR DESCRIPTION
**Pull request by AvitalFineRedis**

### CODE REVIEW

File: src/test/java/redis/clients/jedis/MultiNodePipelineBaseTest.java [hunk 1]
```diff
@@ -0,0 +1,883 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static redis.clients.jedis.Protocol.CLUSTER_HASHSLOTS;
+
+public class MultiNodePipelineBaseTest {
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/MultiNodePipelineBaseTest.java:34]:
```suggestion
public class ClusterPipeliningTest {
```

File: src/main/java/redis/clients/jedis/params/GeoRadiusParam.java [hunk 2]
```diff
@@ -1,21 +1,17 @@
 package redis.clients.jedis.params;
 
-import static redis.clients.jedis.Protocol.Keyword.ASC;
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/params/GeoRadiusParam.java:3]: undo

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/test/java/redis/clients/jedis/ClusterPipeliningTest.java [hunk 3]
```diff
@@ -0,0 +1,883 @@
...
+        Assert.assertEquals(Long.valueOf(union.size()), r9.get());
+        Assert.assertEquals(union, r10.get());
+        Assert.assertTrue(r11.get());
+        Assert.assertTrue(r12.get().get(0) && !r12.get().get(1));
+        Assert.assertEquals(Long.valueOf(1), r13.get());
+        Assert.assertTrue(union.containsAll(r14.get()));
+        Assert.assertEquals(Long.valueOf(2), r15.get());
+        Assert.assertTrue(union.contains(r16.get()));
+        Assert.assertTrue(union.containsAll(r17.get()));
+        Assert.assertEquals(Long.valueOf(1), r18.get());
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/ClusterPipeliningTest.java:271]: [ERROR]   ClusterPipeliningTest.clusterPipelineSet:271 expected:<1> but was:<0>

File: src/test/java/redis/clients/jedis/ClusterPipeliningTest.java [hunk 4]
```diff
@@ -0,0 +1,883 @@
...
+        Response<Long> r9 = p.sunionstore("unionset{|}","myset{|}", "mynewset{|}");
+        Response<Set<String>> r10 = p.smembers("unionset{|}");
+        Response<Boolean> r11 = p.sismember("myset{|}", "foo");
+        Response<List<Boolean>> r12 = p.smismember("myset{|}", "foo", "foobar");
+        Response<Long> r13 = p.srem("myset{|}", "foo");
+        Response<Set<String>> r14 = p.spop("myset{|}", 1);
+        Response<Long> r15 = p.scard("myset{|}");
+        Response<String> r16 = p.srandmember("myset{|}");
+        Response<List<String>> r17 = p.srandmember("myset{|}", 2);
+        Response<Long> r18 = p.smove("myset{|}", "mynewset{|}", "hello");
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/ClusterPipeliningTest.java:251]: Could you please use a better hashtag? E.g. `set`, as in `my{set}`, `mynew{set}`, etc.

File: src/test/java/redis/clients/jedis/ClusterPipeliningTest.java [hunk 5]
```diff
@@ -0,0 +1,883 @@
...
+        Response<Long> r5 = p.bitcount("myotherkey", 1,1);
+        Response<Long> r6 = p.bitpos("mykey{^}", true);
+        Response<Long> r7 = p.bitpos("mykey{^}", false, new BitPosParams(1, 2));
+        Response<List<Long>> r8 = p.bitfield("mynewkey","INCRBY", "i5", "100", "1", "GET", "u4", "0");
+        Response<List<Long>> r9 = p.bitfieldReadonly("hello", "GET", "i8", "17");
+        p.set("myotherkey{^}", "abcdef");
+        Response<Long> r10 = p.bitop(BitOP.AND, "dest{^}", "mykey{^}", "myotherkey{^}");
+        Response<String> r11 = p.get("dest{^}");
+        Response<Boolean> r12 = p.setbit("myotherkey", 7, true);
+        Response<Boolean> r13 = p.getbit("myotherkey", 7);
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/ClusterPipeliningTest.java:498]: Could you please a better hashtag? E.g. `mykey`, as in `{mykey}`, `{mykey}other`, etc.

File: src/test/java/redis/clients/jedis/ClusterPipeliningTest.java [hunk 6]
```diff
@@ -0,0 +1,883 @@
...
+        Response<Long> r10 = p.lrem("mylist", 1, "hello"); // ["foobar"]
+        Response<List<String>> r11 = p.lrange("mylist",0,10);
+        Response<String> r12 = p.rpop("mynewlist{$}"); // ["hello", "hello", "foo"]
+        Response<List<String>> r13 = p.lpop("mylist", 1); // ["foobar"]
+        Response<List<String>> r14 = p.rpop("mynewlist{$}", 2); // ["hello"]
+        Response<Long> r15 = p.linsert("mynewlist{$}", ListPosition.AFTER, "hello", "world"); // ["hello", "world"]
+        Response<Long> r16 = p.lpushx("myotherlist{$}", "foo", "bar");
+        Response<Long> r17 = p.rpushx("myotherlist{$}", "foo", "bar");
+        Response<String> r18 = p.rpoplpush("mynewlist{$}", "myotherlist{$}");
+        Response<String> r19 = p.lmove("mynewlist{$}", "myotherlist{$}", ListDirection.LEFT, ListDirection.RIGHT);
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/ClusterPipeliningTest.java:190]: Could you please use a better hashtag? E.g. `list`, as in `my{list}`, `mynew{list}`, etc.

File: src/main/java/redis/clients/jedis/Protocol.java [hunk 7]
```diff
@@ -253,7 +253,7 @@ private Command() {
 
     AGGREGATE, ALPHA, ASC, BY, DESC, GET, LIMIT, NO, NOSORT, ONE, SET, STORE, WEIGHTS, WITHSCORES,
     RESETSTAT, REWRITE, RESET, FLUSH, EXISTS, LOAD, KILL, LEN, REFCOUNT, ENCODING, IDLETIME, DOCTOR,
-    HELP, FREQ, GETNAME, SETNAME, LIST, MATCH, COUNT, TYPE, UNLOAD, REPLACE, PAUSE, BLOCK, NOACK,
+    HELP, FREQ, GETNAME, SETNAME, LIST, MATCH, COUNT, ANY, TYPE, UNLOAD, REPLACE, PAUSE, BLOCK, NOACK,
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Protocol.java:256]: Put ANY at the list line.

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/Protocol.java (hunk 7 above)
Comment (AvitalFineRedis) [reply src/main/java/redis/clients/jedis/Protocol.java:256]: What is the list line? Do you mean at the end?

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/Protocol.java]: Oops, sorry. "Last" line.

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/Protocol.java]: You've changes it already 😃

**Review by sazzad16 — APPROVED** [review]

**Comment by sazzad16** [comment]
@AvitalFineRedis According to Codecov, this PR is introducing 119 misses. Did you get to check what are those?

**Review by chayim — APPROVED** [review]

**Comment by AvitalFineRedis** [comment]
> @AvitalFineRedis According to Codecov, this PR is introducing 119 misses. Did you get to check what are those?

Yes, I don't think It's my lines.

File: src/main/java/redis/clients/jedis/params/GeoRadiusParam.java [hunk 8]
```diff
@@ -58,6 +60,17 @@ public GeoRadiusParam count(int count) {
     return this;
   }
 
+  public GeoRadiusParam count(int count, boolean any) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/params/GeoRadiusParam.java:63]: This is a new feature. Add proper unit tests for it.

File: src/main/java/redis/clients/jedis/resps/GeoRadiusResponse.java [hunk 9]
```diff
@@ -44,4 +46,20 @@ public long getRawScore() {
   public void setRawScore(long rawScore) {
     this.rawScore = rawScore;
   }
+
+  @Override
+  public boolean equals(Object obj) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/resps/GeoRadiusResponse.java:51]: This is a new feature. Add proper unit tests for it.

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Review by sazzad16 — CHANGES_REQUESTED** [review]
Set 2-space formatter (4 spaces when continuing).

File: src/main/java/redis/clients/jedis/params/GeoRadiusParam.java (hunk 8 above)
Comment (AvitalFineRedis) [reply src/main/java/redis/clients/jedis/params/GeoRadiusParam.java:63]: Added [here](https://github.com/AvitalFineRedis/jedis/pull/1)

File: src/main/java/redis/clients/jedis/resps/GeoRadiusResponse.java (hunk 9 above)
Comment (AvitalFineRedis) [reply src/main/java/redis/clients/jedis/resps/GeoRadiusResponse.java:51]: Added [here](https://github.com/AvitalFineRedis/jedis/pull/1)

File: src/main/java/redis/clients/jedis/params/GeoRadiusParam.java (hunk 8 above)
Comment (AvitalFineRedis) [reply src/main/java/redis/clients/jedis/params/GeoRadiusParam.java:63]: Added [here](https://github.com/AvitalFineRedis/jedis/pull/1)

**Review by dengliming — APPROVED** [review]

**Review by yangbodong22011 — APPROVED** [review]

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**